Feat/multiple instances

Dodecahedr0x opened the pull request.

Dodecahedr0x committed:
Enable multiple instances

Dodecahedr0x committed:
Linting

Dodecahedr0x committed:
Add test CI

Dodecahedr0x committed:
Mostly inexistent tests

Dodecahedr0x committed:
Fix CI

joebuild commented:
This is awesome!!

I want to give the seed/account changes a good look, so will spend some time doing that soon, but overall this is great.

Just a couple minor nits:
- For the `Dao` PDA seeds, we should use a unique string prefix to prevent account collisions. There aren't any collisions here yet but it'll be a good practice to maintain. Please see here for example/convention: https://github.com/joebuild/autocrat/blob/master/programs/autocrat/src/state/mod.rs#L34-L42

- Similarly, let's keep the string prefix as the first part of the seed, in `Proposal` account, i.e. just move the 'dao` seed down in the order.

Dodecahedr0x committed:
Fix seeds

Dodecahedr0x commented:
Fixed!
I also realized I was misusing seeds, cleaned it up.

I'm wondering if  there could be some CU optimization by reducing seeds checks when `has_one` can be used instead
